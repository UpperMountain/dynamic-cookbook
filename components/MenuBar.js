import React from "react";
import { View, StyleSheet, TouchableOpacity, Text } from "react-native";
import { FontAwesome } from "@expo/vector-icons";
var s = require("./Styles");

class MenuBar extends React.Component {
  constructor(props) {
    super();
    this.state = {
      current: 0
    };
  }
  render() {
    const Icon = props => {
      let currentColor =
        props.pos == this.state.current ? "#ffc3a0" : "#757575";
      return (
        <TouchableOpacity
          style={{ alignItems: "center", marginTop: 5 }}
          onPress={() => this.setState({ current: props.pos })}
        >
          <FontAwesome name={props.name} size={26} color={currentColor} />
          <Text style={{ color: currentColor }}>{props.text}</Text>
        </TouchableOpacity>
      );
    };

    const styles = StyleSheet.create({
      bar: {
        position: "absolute",
        bottom: 0,
        borderTopColor: "rgba(108, 123, 138, .24)",
        borderTopWidth: 1,
        width: "100%",
        height: 74,
        backgroundColor: "#fff",
        flexDirection: "row"
      }
    });

    return (
      <View style={[styles.bar, s.shadow]}>
        <Icon pos={0} name={"home"} text={"Home"} />
        <Icon pos={1} name={"th"} text={"Categories"} />
        <Icon pos={2} name={"book"} text={"My Meal"} />
      </View>
    );
  }
}

export default MenuBar;
