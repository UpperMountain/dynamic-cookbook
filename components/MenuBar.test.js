import React from "react";
import MenuBar from "./MenuBar";
import renderer from "react-test-renderer";

it("renders without exploding", () => {
  const cmp = renderer.create(<MenuBar />);
});

it("passes snapshot tests", () => {
  const cmp = renderer.create(<MenuBar />);
  expect(cmp.toJSON()).toMatchSnapshot();
});
