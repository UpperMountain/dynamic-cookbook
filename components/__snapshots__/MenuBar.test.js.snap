// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot tests 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderTopColor": "rgba(108, 123, 138, .24)",
        "borderTopWidth": 1,
        "bottom": 0,
        "flexDirection": "row",
        "height": 74,
        "justifyContent": "space-around",
        "position": "absolute",
        "width": "100%",
      },
      Object {
        "elevation": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 5,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 5,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#ffc3a0",
        }
      }
    >
      Home
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 5,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#757575",
        }
      }
    >
      Categories
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 5,
        "opacity": 1,
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "#757575",
        }
      }
    >
      My Meal
    </Text>
  </View>
</View>
`;
